go_modules/lib/dependabot/go_modules/file_fetcher.rb

```diff
@@ -47,6 +47,10 @@ def go_mod
       def go_sum
         @go_sum ||= fetch_file_if_present("go.sum")
       end
+
+      def recurse_submodules_when_cloning?
+        true
```

Comment:
Something we can try here to mitigate the issue @thepwagner brought up is only install submodules if the initial dependency resolution step failed to run

We do something of that sorts for Terraform modules in https://github.com/dependabot/dependabot-core/blob/95df3a4d92ffc938adc1ece2942a7e9887387573/terraform/lib/dependabot/terraform/file_updater.rb#L242-L253

Reply:
Another option might be we leave the submodule clone by default, but ignore any errors to attempt the update anyway? That way if it didn't matter then we still do the update.